core/schema/container.go

```diff
@@ -122,6 +122,13 @@ func (s *containerSchema) Install() {
 					`environment variables defined in the container (e.g.,
 				"/opt/bin:$PATH").`),
 
+		// NOTE: this is internal-only for now (hidden from codegen via the __ prefix) as we
+		// currently only want to use it for allowing the Go SDK to inherit custom GOPROXY
+		// settings from the engine container. It may be made public in the future with more
+		// refined design.
+		dagql.Func("__withSystemEnvVariable", s.withSystemEnvVariable).
+			Doc(`(Internal-only) Inherit this environment variable from the engine container if set there with a special prefix.`),
```

Comment:
This is more just something to note - the number of "internal-only APIs" has kind of started to grow.

I'd really like to try splitting some of these out, ideally by having a concept of a "privileged" session, which serves those APIs - other calls wouldn't see it. Then codegen wouldn't generate any of these for those.

Reply:
It would be nice to have something more fleshed out here I agree, but this `__` prefix does [result in them being hidden from codegen](https://github.com/sipsma/dagger/blob/1bc2cffabc80e614afea9dc0155a0e77af1aa4c4/dagql/introspection/types.go#L519-L519) (just to be clear, not sure if that's what you were getting at with "Then codegen wouldn't generate any of these for those").